### ISSUE TITLE
Hud element not showing 

### ISSUE THREAD

**Issue by skylakq** [YaLTeR/BunnymodXT#43]
![desktop screenshot 2018 05 28 - 20 39 21 20](https://user-images.githubusercontent.com/30622629/40610841-3e49ab14-62b7-11e8-8a70-c6694bf96953.png)
![desktop screenshot 2018 05 28 - 20 39 49 64](https://user-images.githubusercontent.com/30622629/40610849-4bb0d08e-62b7-11e8-8903-7d538aea6498.png)
![desktop screenshot 2018 05 28 - 20 40 49 27](https://user-images.githubusercontent.com/30622629/40610886-739ad72a-62b7-11e8-9c7c-4fdbebd9ac51.png)
Hud elements do not show!
Same results in and outside of program files (on an ssd anyway not real program files) and on goldsrc package and NGHL
Tried w/steam and without steam (NGHL)
Same result with similar errors.
I tried updating drivers and it worked for one session but then I restart my computer and it's back to this
I have no idea what causes this. 
Tried other stuff to no avail either.

Pastebin for bxt log: https://pastebin.com/azKvBKnb

**Comment by skylakq** [comment]
Oh also bxt and nghl are both up to date

**Comment by YaLTeR** [comment]
For some reason sometimes for some people on some clients BXT doesn't see the client DLL. I'm not quite sure why.

**Comment by skylakq** [comment]
oh ok, don't worry I'll just deal with it

**Comment by skylakq** [comment]
So, after a LOT of troubleshooting I found a solution to get it working consistently on NGHL.
1. Make a batch that only runs rinput and the hl client.
2. Wait for batch to open
3. Manually execute injector.exe 
4. Done!

Don't know why this works and is quite annoying to do over time but at least its functional again. 
Sadly this doesn't work for goldsrc package from what I've tried.
Is this because maybe the injector is loading before half-life has loaded completely hence its not able to hook all the elements?

**Comment by skylakq** [comment]
Yeah tried that just then, exact same result as using the batch file. When i get home later today ill try debugging the sourcecode to see whats wrong

**Comment by ardatelatar3** [comment]
Try using the 6027 Build of half-life it might help.

**Comment by YaLTeR** [comment]
Any news on this?

**Comment by skylakq** [comment]
Nah ive just been workibg around it by loadin the gane seperately and then
loadong the injector

**Comment by RyanDamerell** [comment]
I'm having the same problem, bxt_hud doesn't show up as a command, I'm also getting a lot of errors regarding missing dlls. I downloaded everything through the full goldsrc package (2.3), I'm wondering if the dlls are outdated or something. 

I should also mention, I have the steam version of Half Life and the two gearbox expansions installed.

![ssyCPhTeI4](https://user-images.githubusercontent.com/23424092/72689926-a4731900-3acb-11ea-95ea-43f9303be31c.png)
![XCVWABPrut](https://user-images.githubusercontent.com/23424092/72689927-a50baf80-3acb-11ea-8ce0-0869525d1686.png)

**Comment by RyanDamerell** [comment]
Also just realized, those two screenshots are in reverse chronological order. Whoops.

**Comment by RyanDamerell** [comment]
I'm having the same problem, bxt_hud doesn't even show up as a command, I'm also getting a lot of errors regarding missing dlls. I downloaded everything through the full goldsrc package (2.3), so I'm not sure why that would be, but idk.

Even weirder, hud commands show up in opposing force, but not hl1 or blue shift. At first I thought I was missing dlls, because of the error messages in the debug console, but now I'm not so sure since most of them appear in the debug console even for HLOF. It perplexes me. I actually tried launching op4 and just typing "map c0a0", which mostly worked, but almost all voicelines are missing but the only game-affecting bugs were in blast pit and xen. 
Here's some debug screenshots for hl1 in case that offers any insight.
![XCVWABPrut](https://user-images.githubusercontent.com/23424092/72691562-36365280-3adb-11ea-830b-6b739b4fafb4.png)
![ssyCPhTeI4](https://user-images.githubusercontent.com/23424092/72691567-39c9d980-3adb-11ea-8e33-72ff6948c11f.png)

**Comment by YaLTeR** [comment]
Yeah, doesn't see client for some reason, odd.

**Comment by refrigerators** [comment]
so how could you fix this besides running it through injector?

**Comment by quantumdude836** [comment]
I'm also running into this issue with GoldSrc Package 2.3, but only with the Steam version.

Does it have anything to do with the fact that the Steam version of client.dll is encrypted?

**Comment by Ayman12346** [comment]
![2022-02-07_164747](https://user-images.githubusercontent.com/99199518/152827223-cba98bff-1915-4622-94e9-b4889be58008.png)

**Comment by YaLTeR** [comment]
No need to put this into every issue. Also the BXT updater is made by a different developer and not tracked in this repository.

**Comment by SmileyAG** [comment]
Don't angry on me for answering to old tickets, better to ignore this (I'm cleaning the tracker from solved issues / those that don't have enough details and too old to investigate for both us and ticket owner / have alternative solutions to solve it).

HUD elements are not shown because the client.dll is encrypted and BXT does not decrypt it.
In the one of latest versions of BXT, a check was introduced to exit the version when it detects that the client is encrypted.

For speedruns of the WON category, it is strongly recommended to use game binaries from WON version (1.1.0.7 for HL1 WON, 1.1.0.4 for OP4 WON)
For speedruns of the Steam category, use Steampipe builds from 59xx and newer (e.g. to download older depots use `download_depot` in Steam console or use [DepotDownloader](https://github.com/SteamRE/DepotDownloader), list of depots could be found on [SteamDB](https://steamdb.info/app/70/depots/)).

`GoldSrc Package 2.4` includes both versions of this, so you can speedrun both the WON and Steam categories with that package.
